luis-hurtado/tarea-5/puzzle.js

```diff
@@ -0,0 +1,194 @@
+            item.setAttribute("ondragend", "");
+        }
+        return;
+    })
+}
+const removeDroppable = (items) => {
+    items.forEach((item, i) => {
+        state.emptyBoxIndex = i
+        item.setAttribute("ondrop", "")
+        item.setAttribute("ondragover", "")
+        item.setAttribute("draggable", "false")
+        item.setAttribute("ondragstart", "")
+        item.setAttribute("ondragend", "")
+        item.classList.remove("empty")
+    })
+}
+const setDraggable = (items) => {
+    const [row, col] = getEmptyBox();
+    console.log(state)
+    let left, right, top, bottom = null;
+    if(state.dimension[row][col-1]) left = state.dimension[row][col-1];
+    if(state.dimension[row][col+1]) right = state.dimension[row][col+1];
+
+    if(state.dimension[row-1] !== undefined) top = state.dimension[row-1][col];
+    if(state.dimension[row+1] !== undefined) bottom = state.dimension[row+1][col];
+
+    items.forEach(item => {
+        if(item.innerText == top || 
+            item.innerText == bottom ||
+            item.innerText == right ||
+            item.innerText == left) {
+                item.setAttribute("draggable", "true")
+                item.setAttribute("ondragstart", "dragstart_handler(event)")
+                item.setAttribute("ondragend", "dragend_handler(event)")
+            }
+    })
+}
+
+//setter
+
+const setBoxId = (puzzleBox) =>{
+    for(let i = 0; i < puzzleBox.length; i++){
+        puzzleBox[i].setAttribute("id", `div${i}`)
+    }
+}
+
+/// DRAG HANDLERS
+const dragstart_handler = ev => {
+    console.log("dragstart")
+    ev.dataTransfer.setData("text/plain", ev.target.id)
+    ev.dataTransfer.dropEffect = "move"
+}
+
+const dragover_handler = ev => {
+    console.log("dragOver")
+    ev.preventDefault();    
+}
+
+const drop_handler = ev => {
+    console.log("drag")
+    ev.preventDefault();
+    const data = ev.dataTransfer.getData("text/plain")
+    ev.target.innerText = document.getElementById(data).innerText
+
+    ev.target.classList.remove("empty")
+    ev.target.setAttribute("ondrop", "")
+    ev.target.setAttribute("ondragover", "")
+    document.getElementById(data).innerText = ""
+
+    state.content = getState(puzzleBox)
+    state.dimension = getDimension(state)
+}
+
+function dragend_handler(ev) {
+    console.log("dragEnd")
+    ev.dataTransfer.clearData()
+
+    removeDroppable(document.querySelectorAll('div'))
+
+    setDroppable(document.querySelectorAll('div'))
+    setDraggable(document.querySelectorAll('div'))
+    console.log(state.content)
+    console.log(numbers)
+}
+
+
+
+//shuffle
+const shuffle = (arr) => {
+    const copy = [...arr]
+
+    for(let i = 0; i < copy.length; i++){
+        let j = parseInt(Math.random()*copy.length)
+        let temp = copy[i]
+        copy[i] = copy[j]
+        copy[j] = temp
+    }
+    return copy
+}
+
+//shuffle
+const isSolvable = (arr) => {
+    let numberOfInversion = 0
+
+    for(let i = 0; i < arr.length; i++){
```

Comment:
@luishurtado-globant , you should use modern approach like forEach, map or reduce